packages/date-time-picker/src/vaadin-date-time-picker-mixin.js

```diff
@@ -347,12 +376,49 @@ export const DateTimePickerMixin = (superClass) =>
       this.ariaTarget = this;
     }
 
+    /** @protected */
+    connectedCallback() {
+      super.connectedCallback();
+      document.documentElement.addEventListener('click', this.__onGlobalClick, true);
+    }
+
+    /** @protected */
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      document.documentElement.removeEventListener('click', this.__onGlobalClick, true);
+    }
+
     focus() {
       if (this.__datePicker) {
         this.__datePicker.focus();
       }
     }
 
+    /** @private */
+    __onGlobalClick(event) {
+      const isOpened = this.__datePicker.opened || this.__timePicker.opened;
+      if (!isOpened) {
+        return;
+      }
+
+      const isOutsideClick = event.composedPath().every((node) => {
+        return ![
+          this.__datePicker,
+          this.__datePicker.$.overlay,
+          this.__timePicker,
+          this.__timePicker.$.comboBox.$.overlay,
+        ].includes(node);
+      });
+
+      if (isOutsideClick) {
+        this.__outsideClickInProgress = true;
+
+        setTimeout(() => {
```

Comment:
Could we use `queueMicrotask` here to limit more preciesely how long this flag stays true?

Reply:
It seems that `queueMicrotask` does not provide enough of a duration for the flag to stay true.